pkg/middleware/rbac_test.go

```diff
@@ -0,0 +1,554 @@
+	switch c.Request().Method {
+	case http.MethodGet:
+		return c.JSON(http.StatusOK, "It worked")
+	default:
+		return c.JSON(http.StatusOK, nil)
+	}
+}
+
+func rbacServe(t *testing.T, req *http.Request, resource string, verb client.RbacVerb, skipper echo_middleware.Skipper, rbacAllowed bool, rbacError error, generateIdentity bool) *httptest.ResponseRecorder {
+	var (
+		xrhid string
+		rw    *httptest.ResponseRecorder
+	)
+
+	require.NotNil(t, req)
+
+	if generateIdentity {
+		xrhid = mockXRhUserIdentity(t, "12345", "12345")
+		require.NotEqual(t, "", xrhid)
+	}
+
+	mockRbacClient := mocks_client.NewRbac(t)
+	require.NotNil(t, mockRbacClient)
+	if hasToCallMock(resource, verb, rbacAllowed, rbacError) {
+		mockRbacClient.On("Allowed", xrhid, resource, verb).Return(rbacAllowed, rbacError)
+	}
+
+	e := echo.New()
+	require.NotNil(t, e)
+
+	// Add Rbac middleware
+	e.Use(
+		NewRbac(Rbac{
+			BaseUrl:        config.Get().Clients.RbacBaseUrl,
+			Skipper:        skipper,
+			PermissionsMap: ServicePermissions,
+			Client:         mockRbacClient,
+		}),
+	)
+
+	// Add a handler to avoid 404
+	e.Add(req.Method, req.URL.Path, handleItWorked)
+
+	rw = httptest.NewRecorder()
+	require.NotNil(t, rw)
+
+	if generateIdentity {
+		req.Header.Set(xrhidHeader, xrhid)
+	}
+
+	e.ServeHTTP(rw, req)
+	mockRbacClient.AssertExpectations(t)
+
+	return rw
+}
+
+func skipperTrue(c echo.Context) bool {
+	return true
+}
+
+func skipperFalse(c echo.Context) bool {
+	return false
+}
+
+func TestNewRbacPanics(t *testing.T) {
+	var pm *PermissionsMap = ServicePermissions
+	var client = mocks_client.NewRbac(t)
+	var skipper echo_middleware.Skipper = skipperTrue
+	require.Panics(t, func() {
+		NewRbac(Rbac{
+			BaseUrl:        "http://localhost:8800/api/rbac/v1",
+			Skipper:        skipper,
+			PermissionsMap: nil,
+			Client:         client,
+		})
+	})
+	require.Panics(t, func() {
+		NewRbac(Rbac{
+			BaseUrl:        "http://localhost:8800/api/rbac/v1",
+			Skipper:        skipper,
+			PermissionsMap: pm,
+			Client:         nil,
+		})
+	})
+}
+
+func TestRbacMiddleware(t *testing.T) {
+	type TestCaseGivenRbac struct {
+		Resource string
+		Verb     client.RbacVerb
+		Allowed  bool
+		Err      error
+	}
+	type TestCaseGivenRequest struct {
+		Method string
+		Path   string
+	}
+	type TestCaseGiven struct {
+		GenerateIdentity bool
+		Request          TestCaseGivenRequest
+		MockResponse     TestCaseGivenRbac
+		Skipper          echo_middleware.Skipper
+	}
+	type TestCaseExpected struct {
+		Code int
+		Body string
+	}
+	type TestCase struct {
+		Name     string
+		Given    TestCaseGiven
+		Expected TestCaseExpected
+	}
+
+	testPath := "/api/content-sources/v1"
+
+	testCases := []TestCase{
+		{
+			Name: "Skipper return true",
+			Given: TestCaseGiven{
+				GenerateIdentity: false,
+				Request: TestCaseGivenRequest{
+					Method: http.MethodGet,
+					Path:   testPath + "/repositories/",
+				},
+				MockResponse: TestCaseGivenRbac{
+					// Resource: "repositories",
```

Comment:
Why is this commented out here?

Reply:
thanks mate! This is commented out because the mocked interface is not called for that test case; If I fill the information, it is considered that should be called, the mocked interface is prepared, and the test execution was failing because no call was received on the mocked instance.

I have added a note about it and cleaned-up the comments.